### PR TITLE
Enhancement Parse Date for support format name

### PR DESCRIPTION
**Pull request by NahuFigueroa97**
|Related issue|
|---|
|#22143|

## Summary

This pull request addresses three issues:

1. **Interpretation of Date Format Names:**
   - Previously, date format names were not being interpreted correctly in decoders.
   - The solution involves verifying that the format name is in the list of supported formats and selecting the corresponding format. This required changes in all decoders using the format instead of the name.

2. **Installation Path of tzdb:**
   - The tzdb was being installed in /root/Downloads.
   - The fix involves utilizing the functionality provided by the API to set the installation path. The path is no longer set at the time of download with CPM; instead, the installation path is specified at runtime.

3. **Improved Error Handling for Undefined Locale:**
   - Previously, when the locale was not defined, the error message was challenging to follow.
   - The error message for an undefined locale has been improved to provide better clarity.

## Changes

- Modified decoders to correctly interpret date format names.
- Updated tzdb installation path to use API functionality, allowing runtime specification of the installation path.
- Improved error message for cases where the locale is not defined.

## Implementation Details

- For the interpretation of date format names, changes were made in the logic of each affected decoder. The verification of the format name against the supported list was added.

- The tzdb installation path was updated to use the API's functionality for specifying the installation path at runtime. The changes were applied consistently across the relevant codebase.

- The error handling for undefined locales was improved by adjusting the error message to provide more helpful information.

### CODE REVIEW

File: src/engine/source/cmds/src/start.cpp [hunk 1]
```diff
@@ -138,6 +142,22 @@ void runStart(ConfHandler confManager)
...
+            auto& tzdb  = date::reload_tzdb();
+        }
+        catch (const std::exception& e)
+        {
+            LOG_ERROR("An error occurred while reload the tzdb: {}.", utils::getExceptionStack(e));
+            exitHandler.execute();
+            return;
+        }
+    }
+
```
Comment (juliancnn) [inline src/engine/source/cmds/src/start.cpp:160]: This block is part of the engine initialization, it should be inside the try, just before or inside the HLP initialization, and it should throw a runtime_exception describing the error without the return.

File: src/engine/source/hlp/src/parsers/date.cpp [hunk 2]
```diff
@@ -89,8 +89,8 @@ SemParser getSemParser(std::string_view targetField,
  * Supported formats, this will be injected by the config module in due time
  */
 const std::vector<std::tuple<std::string, std::string>> TIME_FORMAT {
-    {"ANSIC", "%a %b %d %T %Y"},        // Mon Jan _2 15:04:05 2006
-    {"UnixDate", "%a %b %d %T %Z %Y"},  // Mon Jan _2 15:04:05 MST 2006
+    {"ANSIC", "%a %b %d %T %Y"},        // Mon Jan 2 15:04:05 2006
+    {"UnixDate", "%a %b %d %T %Z %Y"},  // Mon Jan 2 15:04:05 MST 2006
```
Comment (juliancnn) [inline src/engine/source/hlp/src/parsers/date.cpp:93]: Do not delete the `_`, this marks the optional., %d is the day of the month as a decimal number. The modified command %Nd where N is a positive decimal integer specifies the maximum number of characters to read. If not specified, the default is 2. Leading zeroes are permitted but not required.

**Review by juliancnn — CHANGES_REQUESTED** [review]

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**